src/test_workflow/smoke_test/smoke_test_runner_opensearch.py

```diff
@@ -28,13 +28,26 @@ def __init__(self, args: TestArgs, test_manifest: TestManifest) -> None:
         super().__init__(args, test_manifest)
         logging.info("Entering Smoke test for OpenSearch Bundle.")
 
-        # TODO: Download the spec from https://github.com/opensearch-project/opensearch-api-specification/releases/download/main-latest/opensearch-openapi.yaml
-        spec_file = os.path.join(os.path.dirname(os.path.abspath(__file__)), "smoke_tests_spec", "opensearch-openapi.yaml")
-        self.spec_ = Spec.from_file_path(spec_file)
+        # Below URL is for the pre-release latest. In the future may consider use formal released spec.
+        self.spec_url = "https://github.com/opensearch-project/opensearch-api-specification/releases/download/main-latest/opensearch-openapi.yaml"
+        self.spec_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), "smoke_tests_spec", "opensearch-openapi.yaml")
+        self.download_spec(self.spec_url, self.spec_path)
+        self.spec_ = Spec.from_file_path(self.spec_path)
         self.mimetype = {
             "Content-Type": "application/json"
         }
-        # self.openapi = openapi_core.OpenAPI.from_file_path(spec_file)
+
+    def download_spec(self, url: str, local_path: str) -> None:
+        try:
+            response = requests.get(url, timeout=10)
+            if response.status_code == 200:
+                with open(local_path, "wb") as file:
+                    file.write(response.content)
```

Comment:
Suggest using a temp dir `work.dir` to store the new file instead of overriding the existing local copy.

Thanks.

Reply:
Updated. Renamed the local spec file so now new downloaded spec won't override the local one. Thanks.